### PR TITLE
[RW-3459] [risk=no] Change 'data set' to 'dataset' in user facing texts

### PR DESCRIPTION
**Pull request by andy7i**
Change 'data set' to 'dataset' in user facing texts such as tiles, help text, and call to actions

Changed
data set -> dataset
Data Set -> Dataset
Data set -> Dataset

Did not change the following variations since they are not user-facing and it is outside of the scope of this ticket as I understand it (further discussion in [RW-3459](https://precisionmedicineinitiative.atlassian.net/browse/RW-3459)):
dataSet
data_set
data-set

Example screenshots:
Screen Shot 2019-10-07 at 4 38 11 PM
Screen Shot 2019-10-07 at 4 38 25 PM

Verified that rainforest tests are updated to 'dataset' rather than 'data set'.

### CODE REVIEW

**Comment by blrubenstein** [comment]
@andy7i can you file a tech-debt: correctness ticket for changing all forms to `dataset` when you get the chance (including api variables and DB columns?)

**Comment by andy7i** [comment]
Based on discussions on tech debt priorities, backend changes were put into a new ticket: https://precisionmedicineinitiative.atlassian.net/browse/RW-3459

File: ui/src/app/components/resource-card.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ const resourceTypeStyles = reactStyles({
   conceptSet: {
     backgroundColor: colors.resourceCardHighlights.conceptSet
   },
-  dataSet: {
+  dataset: {
```
[the remark this answers is not in the document]
Comment (andy7i) [reply ui/src/app/components/resource-card.tsx]: Right! I meant to add a comment about this in the PR description. I think that after I put it in the commit description, I forgot to add it to the PR.

File: ui/src/app/components/resource-card.tsx [hunk 2]
```diff
@@ -483,14 +483,14 @@ export class ResourceCard extends React.Component<Props, State> {
         <ModalBody>
           <div style={{paddingBottom: '1rem'}}>
             The {this.resourceType} <b>{fp.startCase(this.displayName)}&nbsp;</b>
-            is referenced by the following Data Sets:
+            is referenced by the following Datasets:
```
[the remark this answers is not in the document]
Comment (andy7i) [reply ui/src/app/components/resource-card.tsx]: It looks like we also capitalize Cohort and Notebook in certain places. For example:

Screen Shot 2019-10-10 at 11 15 05 AM

and

```Build a dataset by selecting the
            variables and values for one or more of your cohorts. Then export the completed Dataset
            to Notebooks where you can perform your analysis```

We aren't consistent in this text with capitalization of "dataset" either. I'll make those changes.

File: ui/src/app/components/resource-card.tsx [hunk 3]
```diff
@@ -483,14 +483,14 @@ export class ResourceCard extends React.Component<Props, State> {
         <ModalBody>
           <div style={{paddingBottom: '1rem'}}>
             The {this.resourceType} <b>{fp.startCase(this.displayName)}&nbsp;</b>
```
[the remark this answers is not in the document]
Comment (andy7i) [reply ui/src/app/components/resource-card.tsx]: I started updating these but realized that this refactor is a bit larger than expected so I'm pulling that into a separate PR.

File: ui/src/app/components/resources.tsx [hunk 4]
```diff
@@ -62,13 +62,13 @@ export class ResourceCardMenu extends React.Component<ResourceCardMenuProps> {
               </MenuItem>
             </React.Fragment>;
           }],
-          ['dataSet', () => {
+          ['dataset', () => {
```
[the remark this answers is not in the document]
Comment (andy7i) [reply ui/src/app/components/resources.tsx]: makes sense to me. Updated.

File: ui/src/app/utils/resourceActions.ts [hunk 5]
```diff
@@ -13,15 +13,15 @@ export enum ResourceType {
...
   COHORT_REVIEW = 'cohortReview',
   CONCEPT_SET = 'conceptSet',
-  DATA_SET = 'dataSet',
+  DATA_SET = 'dataset',
   INVALID = 'invalid'
 }
 
 export const ResourceTypeDisplayNames = new Map()
   .set(ResourceType.NOTEBOOK, 'notebook')
   .set(ResourceType.COHORT, 'cohort')
```
[the remark this answers is not in the document]
Comment (andy7i) [reply ui/src/app/utils/resourceActions.ts]: good catch

**Pull request merged**